engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1695,6 +1607,42 @@ public void run()
             }
             endWindowStatsMap.put(shb.getNodeId(), endWindowStats);
 
+            if (!oper.getInputs().isEmpty()) {
+              long latency = Long.MAX_VALUE;
```

Comment:
@gauravgopi123 The slowest upstream operator should be updated upon the next stats update, correct?